src/botservice/azext_bot/custom.py

```diff
@@ -0,0 +1,466 @@
+        'tenantId' : profile.get_subscription(subscription=client.config.subscription_id)['tenantId'],
+        'subscriptionId' : client.config.subscription_id
+    }
+
+def get_bot(cmd, client, resource_group_name, resource_name, bot_json = None):
+    raw_bot_properties = client.bots.get(
+        resource_group_name = resource_group_name,
+        resource_name = resource_name
+    )
+    if bot_json:
+        return create_bot_json(cmd, client, resource_group_name, resource_name, raw_bot_properties=raw_bot_properties)
+
+    return raw_bot_properties
+
+def create_app(cmd, client,resource_group_name, resource_name, description, kind, appid, password,storageAccountName, location, sku, appInsightsLocation, bot_json, language):
+    if kind == 'function':
+        template_name = 'functionapp.template.json'
+        if language == 'Csharp':
+            zip_url = 'https://connectorprod.blob.core.windows.net/bot-packages/csharp-abs-functions_emptybot.zip'
+        else:
+            zip_url = 'https://connectorprod.blob.core.windows.net/bot-packages/node.js-abs-functions_emptybot_funcpack.zip'
+
+    else:
+        kind = 'sdk'
+        template_name = 'webapp.template.json'
+        if language == 'Csharp':
+            zip_url = 'https://connectorprod.blob.core.windows.net/bot-packages/csharp-abs-webapp_simpleechobot_precompiled.zip'
+        else:
+            zip_url = 'https://connectorprod.blob.core.windows.net/bot-packages/node.js-abs-webapp_hello-chatconnector.zip'
+
+    create_new_storage = False
+    if not storageAccountName:
+        create_new_storage = True
+        import re
+        import string
+        import random
+        storageAccountName = re.sub(r'[^a-z0-9]','',resource_name[:10] + ''.join(random.choice(string.ascii_lowercase+string.digits) for _ in range(4)))
+
+    paramsdict = {
+    "location" : location,
+    "kind" : kind,
+    "sku" : sku,
+    "siteName" : resource_name,
+    "description" : description,
+    "appId" : appid,
+    "appSecret" : password,
+    "storageAccountResourceId" : "",
+    "serverFarmId" : "/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Web/serverfarms/{2}".format(client.config.subscription_id, resource_group_name, resource_name),
+    "zipUrl" : zip_url,
+    "createNewStorage" : create_new_storage,
+    "storageAccountName" : storageAccountName,
+    "botEnv" : "prod",
+    "useAppInsights" : True,
+    "appInsightsLocation" : appInsightsLocation,
+    "createServerFarm" : True,
+    "serverFarmLocation": location.lower().replace(' ', ''),
+    "azureWebJobsBotFrameworkDirectLineSecret" : "",
+    "botId" : resource_name
+    }
+    params = {k: {'value' : v} for k,v in paramsdict.items()}
+
+    import os
+    import json
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    deploy_result = deploy_arm_template(
+        cmd = cmd,
+        resource_group_name = resource_group_name,
+        template_file = '{0}\\{1}'.format(dir_path, template_name),
```

Comment:
Will this work on Linux/macOS?

Reply:
changed to use os.path.join instead of string concatenation. That will make it os independent.